src/probability/integration.lean

```diff
@@ -235,4 +235,27 @@ begin
   ring
 end
 
+/-- Independence of functions `f` and `g` into arbitrary types is characterized by the relation
+  `E[(φ ∘ f) * (ψ ∘ g)] = E[φ ∘ f] * E[ψ ∘ g]` for all measurable `φ` and `ψ` with values in `ℝ`
+  satisfying appropriate integrability conditions. -/
+theorem indep_fun_iff_integral_comp_mul [is_finite_measure μ]
+  {β β' : Type*} {mβ : measurable_space β} {mβ' : measurable_space β'}
+  {f : α → β} {g : α → β'} {hfm : measurable f} {hgm : measurable g} :
+  indep_fun f g μ ↔
+  ∀ {φ : β → ℝ} {ψ : β' → ℝ},
+    measurable φ → measurable ψ → integrable (φ ∘ f) μ → integrable (ψ ∘ g) μ →
+    integral μ ((φ ∘ f) * (ψ ∘ g)) = integral μ (φ ∘ f) * integral μ (ψ ∘ g) :=
+begin
+  have I : ∀ {A : set α} (hA : measurable_set A), integral μ (A.indicator 1) = (μ A).to_real,
+    from λ _ hA, (integral_indicator_const (1 : ℝ) hA).trans ((smul_eq_mul _).trans (mul_one _)),
```

Comment:
It's maybe worth having this as a specialized lemma, because I imagine we will use it quite often.

Reply:
Ah, why not. Somewhere with `set_integral_const`, `integral_indicator_const` and the like. I will do that then.

Reply:
Done :-)